### PR TITLE
Add Palemoon support

### LINKED ISSUE #112: xkit support on pale moon
Pale Moon is a web browser based on firefox, but trying to install the new xkit brings back "not available for firefox 24.9." Is there a workaround or fix for this?

### PR DESCRIPTION
**Pull request by Wolvan**
A new task has been introduced to gulp that allows building an extension
compatible with the Palemoon browser. Palemoon extensions need to be
built using Mozilla's old CFX tool, which unfortunately requires Python
2.7 to be installed. This is also the main reason why `build:palemoon`
is not part of the `build` task.
The source to built from is the Firefox extension, since there are no
incompatibilities in the Firefox code we use, it was just the build
tool being incompatible.
Closes #112

### CODE REVIEW

**Comment by Ordinaryperson2** [comment]
I just built this and it works great! Thanks for your work!

**Comment by 0xazure** [comment]
I think there needs to be a larger discussion around whether we want to commit to officially supporting Pale Moon before a PR like this can be accepted.

@new-xkit/admin we're going to have to discuss this as a team, I think.

The biggest question is around numbers: is this actually something that a _significant portion_ of our userbase is interested in?  Taking on maintenance of yet another browser is no small task.  The maintenance burden needs to be evaluated, for example:

- how different is the Pale Moon addon structure from the current `jpm`-based Firefox one?
- what is Pale Moon's addon publishing and updating apparatus?
- how much extra maintenance will this add to our builds toolchain?

I would personally be interested in waiting until we have results from something like #730 before going ahead.

**Comment by Wolvan** [comment]
What does #730 have to do with this?

**Comment by Wolvan** [comment]
>how different is the Pale Moon addon structure from the current jpm-based Firefox one?

I have not looked at the specifices, but building does require the index.js to be in another folder and a few additions/deletions from the package.json (which the gulp script does automatically)
>what is Pale Moon's addon publishing and updating apparatus?

Same as our current firefox updating and publishing: A release entry on our /releases page. I disabled updates for this version of Pale Moon though, since I decided to work on the update.rdf if/once we build pm officially

>how much extra maintenance will this add to our builds toolchain?

basically none. It builds perfectly fine with what we have for Firefox since we are not using any Firefox' GUI Apis, everything we use is compatible to old firefox/palemoon

**Comment by 0xazure** [comment]
> What does #730 have to do with this?

#730 is the proposal for a user survey, where we could, presumably, gather information related to our userbase's browser usage to give us hard numbers on the percentage of XKit users who also use Pale Moon.  It would inform our decision as far as whether or not Pale Moon support is something the userbase wants.

**Comment by Wolvan** [comment]
Fair enough, I see what you mean. We could also just post a poll on our blog for the time being before we do a full survey. Since, quite honestly, I don't believe #730 being something that happens in the near future

**Comment by Wolvan** [comment]
Closing because this is not discussed properly
